application/common/core.cpp

```diff
@@ -43,19 +44,61 @@ void loop()
 
     // Get the stage as it will receive most events.
     Loom2D::Stage *stage = Loom2D::Stage::smMainStage;
-    
+
+    if(gPendingResizeEvent && stage)
+    {
+        // Fire a resize event. We do this at startup so apps can size them
+        // selves properly before first render.
+        int winWidth, winHeight;
+        SDL_GetWindowSize(gSDLWindow, &winWidth, &winHeight);
+        SDL_GL_GetDrawableSize(gSDLWindow, &winWidth, &winHeight);
+        stage->noteNativeSize(winWidth, winHeight);
+        GFX::Graphics::setNativeSize(winWidth, winHeight);
+
+        gPendingResizeEvent = false;
+    }
+
     /* Check for events */
     while (SDL_PollEvent(&event))
     {
         if (event.type == SDL_QUIT)
         {
-            done = 1;
+            // Terminate execution.
+            gLoomExecutionDone = 1;
+            continue;
         }
 
         // Bail on the rest if no stage!
         if(!stage)
             continue;
 
+        // Adjust coordinates for mouse events to work proprly on high dpi screens.
+        if(event.type == SDL_MOUSEMOTION 
+            || event.type == SDL_MOUSEBUTTONDOWN 
+            || event.type == SDL_MOUSEBUTTONUP)
+        {
+            if (SDL_GetWindowFlags(gSDLWindow) & SDL_WINDOW_ALLOW_HIGHDPI)
+            {
+                // We work in drawable space but OS gives us these events in 
+                // window coords - so scale. Usually it's an integer scale.
```

Comment:
Usually? Should we be using floating point division instead of integer division for the ratios below or is it pretty certain that int div won't cause problems?